### PR TITLE
[1184] Add the edit page for training routes

### PR DESCRIPTION
**Pull request by stellenberger**
### Context

Added an edit page for training routes. This is a pre-requisite to the trello card [1127] Add route indicator

### Changes proposed in this pull request

- Changed the title of both new and edit page to `What type of training are they doing?`
- Added an edit page to edit the training route they are on
- Added a new controller `TrainingRouteController` + routes
- Said controller has edit/update functionality only 
- Said controller will validate that the user is a draft before proceeding

### Guidance to review

- On creation of new trainee, you will see the new trainee title
- Once you create a new trainee (or view a trainee draft), use the slug of that trainee and navigate to: `trainees/{traineeslug}/training-routes/edit` to see the edit page

### Screenshots

- `https://localhost:5000/trainees/{traineeslug}/training-routes/edit`

Screenshot 2021-03-02 at 15 50 39

### CODE REVIEW

**Comment by stellenberger** [comment]
[[S] Add the edit route page](https://trello.com/c/EE2fXgZ8/1184-s-add-the-edit-route-page)

**Review by edwardhorsford — APPROVED** [review]

File: app/views/trainees/edit.html.erb [hunk 1]
```diff
@@ -0,0 +1,28 @@
+<%= render PageTitle::View.new(title: "trainees.edit", has_errors: @trainee.errors.present?) %>
+
+<%= content_for(:breadcrumbs) do %>
+  <%= render GovukComponent::BackLink.new(
+    text: "Back",
+    href: trainees_path
```
Comment (AbigailMcP) [inline app/views/trainees/edit.html.erb:6]: I think this page will eventually be accessed from the `/review-draft` or `/check-details` pages. In that case, this should probably be a `DynamicBackLink` component in preparation.

File: app/controllers/trainees_controller.rb [hunk 2]
```diff
@@ -47,7 +47,11 @@ def create
   def update
     authorize trainee
     trainee.update!(trainee_params)
-    redirect_to page_tracker.last_origin_page_path
+    redirect_to trainee
+  end
+
+  def edit
```
Comment (AbigailMcP) [inline app/controllers/trainees_controller.rb:53]: We need a guard here against editing a non-draft trainee. I can access the route manually in the URL with a non-draft trainee and update their `training_route` which I believe is a no-no!

cc. @edwardhorsford

File: app/controllers/trainees_controller.rb [hunk 3]
```diff
@@ -47,7 +47,11 @@ def create
   def update
```
Comment (AbigailMcP) [inline app/controllers/trainees_controller.rb:47]: Same here re guarding against updating a non-draft trainee's `training_route`

**Review by AbigailMcP — COMMENTED** [review]
Just a couple of comments! Am I right in thinking this new page will be access from this link on `/review-draft` and `/check-details`:

Screenshot 2021-03-02 at 16 44 00

And is there a separate card for adding that?

File: app/controllers/trainees/training_routes_controller.rb [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+    def update
+      authorize trainee
+      trainee.update!(trainee_params)
+      redirect_to trainee
+    end
+
+  private
+
+    def trainee_params
+      params.fetch(:trainee, {}).permit(:training_route)
```
Comment (AbigailMcP) [inline app/controllers/trainees/training_routes_controller.rb:20]: You should always have a trainee param:

```suggestion
      params.require(:trainee).permit(:training_route)
```

**Review by AbigailMcP — APPROVED** [review]

**Pull request merged**